First Pull Request

jishaGithub opened the pull request.

jishaGithub committed:
Add files via upload

jishaGithub committed:
Add CustomerAuthenticationService

jishaGithub committed:
Merge pull request #2 from jishaGithub/jishaGithub-patch-2

Add CustomerAuthenticationService

lucasz-student commented:
Everything looks good; it's very organized.

lucasz-student reviewed (approved):
Looks Good

jishaGithub committed:
Update MySQLBranchDAO.java

jishaGithub committed:
Update MySQLBranchDAO.java

jishaGithub committed:
Update Branch.java

jishaGithub committed:
Update CustomerAuthentication.java

jishaGithub committed:
Update PromotionService.java

jishaGithub committed:
Update VehicleFeatureService.java

jishaGithub committed:
Update CustomerAuthentication.java

jishaGithub committed:
Merge pull request #3 from jishaGithub/jishaGithub-patch-2

Jisha GitHub patch 2

jishaGithub commented:
> Looks Good

Thank you

jishaGithub commented:
> Structure: 1- Inside dao package you have another package MySQLDAO , AbstractDAO class and GenericDAO interface. 2- In my opinion, you should only have one package – the dao one. Why is the abstract class and interface separately? Also, your AbstractDAO implements GenericDAO. And your dao classes extend AbstractDAO. Why these classes are not implementing directly GenericDao?? So, this abstract class here with this usage is pointless in my opinion. Also, your config file inside dao can be part of another package. Naming of dao classes : it could be simplier. We know that you are using MySQL from the db.properties file. I liked the rest, you kept it simple and useful :)

Hello Adela,

Thank you for your feedback. 

One of the requirements for the homework was to create DAO classes with interfaces, abstract classes and generics. That's why I implemented it like that. You are correct, I didn't use abstractDAO properly, therefore I have updated the abstractDAO and now abstractDAO loads the configuration file as well as gets the connection for each DAO class. This helped me to reduce the number of lines in each of my DAO class and I was able to reuse my code in the abstractDAO for each of its DAO subclasses. 
I will try to move my config file to a different package. 
I created a new package called MYSQLDAO inside my DAO package because I think that way it will be much easier to extend this application to work for another databases.

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/Main.java:
```diff
@@ -0,0 +1,57 @@
+        BranchService branchService = new BranchService(new MySQLBranchDAO());
+        branchService.findById(3);
+        branchService.selectAll();
+        //branchService.newRow(new Branch("Branch H",2));
```
No comments in code.

jishaGithub replied:
deleted all the comments

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/Main.java:
```diff
@@ -0,0 +1,57 @@
+        CustomerService customerService = new CustomerService(new MySQLCustomerDAO());
+        customerService.findById(3);
+        customerService.selectAll();
+        CustomerAuthenticationService customerAuthenticationService = new CustomerAuthenticationService(new MySQLCustomerAuthenticationDAO());
+        customerAuthenticationService.selectAll();
+        customerAuthenticationService.updateRow(new CustomerAuthentication("jd2023","5634",4),4);
+        InsuranceService insuranceService = new InsuranceService(new MySQLInsuranceDAO());
+        //insuranceService.newRow(new Insurance("Basic Insurance", 100.0,5));
```
Test comment

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/db/dao/GenericDAO.java:
```diff
@@ -0,0 +1,11 @@
+
+import java.util.ArrayList;
+
+public interface GenericDAO<T> {
```
Do we really need this interface?

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/db/dao/GenericDAO.java:
```diff
@@ -0,0 +1,11 @@
+public interface GenericDAO<T> {
+    Boolean findById(int id);
+    ArrayList<T> selectAll();
+    void addNewRow(T t);
```
can be just add

jishaGithub replied:
done

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/db/dao/GenericDAO.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.ArrayList;
+
+public interface GenericDAO<T> {
+    Boolean findById(int id);
+    ArrayList<T> selectAll();
```
```suggestion
    T get(int id);
    ArrayList<T> get();
```

jishaGithub replied:
Done

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchDAO.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    @Override
+    public Boolean findById(int id1) {
```
```suggestion
    public Boolean findById(int id) {
```

jishaGithub replied:
Fixed this issue

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchDAO.java:
```diff
@@ -0,0 +1,112 @@
+                String branchName = result.getString("branch_name");
+                int branchLocationId = result.getInt("branch_location_id");
+                Branch branch = new Branch(branchId, branchName, branchLocationId);
+                System.out.println(branch);
```
use Logger here

jishaGithub replied:
Done

VbKoshak commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchDAO.java:
```diff
@@ -0,0 +1,112 @@
+    public MySQLBranchDAO() {
+        ConfigFileDAO.loadPropertyConfigFile();
+    }
+
```
`private final static String GET_BY_ID = "SELECT * FROM branch where id=?";`

`PreparedStatement preparedStatement = connection.prepareStatement(GET_BY_ID)`;

jishaGithub replied:
Done

VbKoshak commented on DbConnection/pom.xml:
```diff
@@ -0,0 +1,28 @@
+            <version>2.9.0</version>
+        </dependency>
+    </dependencies>
+</project>
```
add an empty line in the end of file

jishaGithub replied:
Done

jishaGithub committed:
Updated AbstractDAO.java

I updated AbstractDAO class. The AbstractDAO class now includes all the connection information and just loads the db configuration files for all the dao classes.

jishaGithub committed:
Update pom.xml

jishaGithub committed:
Update Main.java

jishaGithub committed:
Add files via upload

jishaGithub committed:
Changed all DAO classes

jishaGithub committed:
Update MySQLVehicleHasVehicleFeatureDAO.java

jishaGithub committed:
Update MySQLVehicleFeatureDAO.java

jishaGithub committed:
Update MySQLVehicleDAO.java

jishaGithub committed:
Update MySQLVehicleCategoryDAO.java

jishaGithub committed:
Update MySQLRentalRateDAO.java

jishaGithub committed:
Update MySQLRentalDAO.java

jishaGithub committed:
Update MySQLPromotionDAO.java

jishaGithub committed:
Update MySQLPaymentMethodDAO.java

jishaGithub committed:
Update MySQLPaymentDAO.java

jishaGithub committed:
Update MySQLInsuranceDAO.java

jishaGithub committed:
Update MySQLEmployeeDAO.java

jishaGithub committed:
Update MySQLBranchLocationDAO.java

jishaGithub committed:
Changes to pom.xml

added dependency for log4j2

jishaGithub committed:
changes to configDAO

Moved ConfigFileDAO.java to a new folder called connection.

jishaGithub committed:
Delete ConfigFileDAO.java

jishaGithub committed:
Update db.properties

jishaGithub committed:
Update db.properties

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/Main.java:
```diff
@@ -0,0 +1,55 @@
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
```
```suggestion
```

jishaGithub replied:
Done!

>

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/connection/ConfigFileDAO.java:
```diff
@@ -0,0 +1,30 @@
+            dataSource.setMaxTotal(10);
+            dataSource.setMaxIdle(10);
+        } catch (IOException e) {
+            System.out.println("Error loading the property file");
```
Use logger instead of printing to console

jishaGithub replied:
Done

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/GenericDAO.java:
```diff
@@ -0,0 +1,11 @@
+package com.solvd.laba.db.dao;
```
Is it possible to just put the abstract methods in this interface inside the abstract class and just delete this interface?

jishaGithub replied:
I made some changes. Now my interface includes some methods and are implemented by all the DAO classes as well as all the service classes. The abstract class is providing common functionality to all the dao classes.

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/model/Rental.java:
```diff
@@ -0,0 +1,127 @@
+    private int promotionId;
+    private int employeeId;
+
+    public Rental(int id, Date rentalDate, Date returnDate, int customerId, int insuranceId,
```
This may not be wrong, but why are there two constructors in every model class?

jishaGithub replied:
Yeah I created an extra custom constructor without an id as a parameter because my database primary key is set to auto increment. I was able to update the table with that constructor. I just realized after your comment, that it is unnecessary so I removed all those extra constructors.

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/service/DAOService.java:
```diff
@@ -0,0 +1,9 @@
+package com.solvd.laba.db.service;
```
Isn't this the same as the genericDAO

jishaGithub replied:
I have made changes to the code

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/service/MySQLService/BranchLocationService.java:
```diff
@@ -0,0 +1,38 @@
+package com.solvd.laba.db.service.MySQLService;
```
Aren't your service layers the same as the DAO classes? I remember the purpose of service layers is to use multiple methods from the DAO simultaneously, or change the methods slightly.

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchDAO.java:
```diff
@@ -0,0 +1,101 @@
+                logger.info("No record found. Invalid ID");
+            }
+        } catch (SQLException e) {
+            logger.info("SQL query error");
```
Maybe these should be logger.error() level messages instead of logger.info()

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchLocationDAO.java:
```diff
@@ -0,0 +1,105 @@
+                logger.info("No record found. Invalid ID");
+            }
+        } catch (SQLException e) {
+            logger.info("SQL query error:"+e.getMessage());
```
You don't want the stack trace?

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchDAO.java:
```diff
@@ -0,0 +1,101 @@
+                String branchName = result.getString("branch_name");
+                int branchLocationId = result.getInt("branch_location_id");
+                return new Branch(branchId, branchName, branchLocationId);
+            } else {
```
You don't actually need the else statement; you can just put the body of your else statement after the body of the if statement and the catch blocks. If the code works as expected, the model gets returned, but if it doesn't, it just goes to whatever else is in the body of the method.

jishaGithub replied:
You are right. Thanks for the tip. I have updated my code.

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLBranchLocationDAO.java:
```diff
@@ -0,0 +1,105 @@
+            preparedStatement.executeUpdate();
+            logger.info("Insertion complete");
+        } catch (SQLException e) {
+            logger.info("Error executing SQL query:"+e.getMessage());
```
stack trace? Same with the other errors

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLCustomerAuthenticationDAO.java:
```diff
@@ -0,0 +1,99 @@
+            }
+        } catch (SQLException e) {
+            logger.info("Error executing SQL query:");
+            logger.error(e.getMessage(), e);
```
There's logger.error() here but why is the next exception not logger.error(), but logger.info()?

jishaGithub replied:
Updated the code

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO/MySQLCustomerDAO.java:
```diff
@@ -0,0 +1,104 @@
+            preparedStatement.executeUpdate();
+            logger.info("Insertion complete");
+        } catch (SQLException e) {
+            logger.info("Error:"+e.getMessage());
```
Maybe make all the error Texts the same in all DAO classes

jishaGithub replied:
Done

lucasz-student commented on DbConnection/src/main/java/com/solvd/laba/db/model/Vehicle.java:
```diff
@@ -0,0 +1,124 @@
+    private int year;
+    private String color;
+    private int mileage;
+    private String licenseNumber;
```
Is license number a string; I'm probably wrong I'm not sure

jishaGithub replied:
Yes, some license number can contains alphabets as well and also we wont be using license number for any arithmetic operations. So that's why I used string instead of long/int.

jishaGithub committed:
Update Main.java

jishaGithub committed:
ConfigFileDAO changes added wait time

jishaGithub committed:
Added logger.info

jishaGithub committed:
Delete DbConnection/src/main/java/com/solvd/laba/db/connection directory

jishaGithub committed:
Added util package

jishaGithub committed:
Add files via upload

jishaGithub committed:
Delete DbConnection/src/main/java/com/solvd/laba/db/dao/MySQLDAO directory

jishaGithub committed:
Delete GenericDAO.java

jishaGithub committed:
Delete DbConnection/src/main/java/com/solvd/laba/db/service directory

jishaGithub committed:
Updates to service layer

jishaGithub committed:
Changes to Main

jishaGithub committed:
Update Branch.java

jishaGithub committed:
Update BranchLocation.java

jishaGithub committed:
Update Customer.java

jishaGithub committed:
Update CustomerAuthentication.java

jishaGithub committed:
Update Employee.java

jishaGithub committed:
Update Insurance.java

jishaGithub committed:
Update Payment.java

jishaGithub committed:
Update Promotion.java

jishaGithub committed:
Update Rental.java

jishaGithub committed:
Update Vehicle.java

jishaGithub committed:
Update VehicleCategory.java

jishaGithub committed:
Update VehicleFeature.java

jishaGithub committed:
Changes to PromotionDAO

adelagega commented on DbConnection/src/main/java/com/solvd/laba/db/util/ConfigFileDAO.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.Properties;
+
+public class ConfigFileDAO {
+    private static final Logger logger = LogManager.getLogger(BranchDAO.class);
```
incorrect logger class

jishaGithub replied:
Done

adelagega commented on DbConnection/src/main/java/com/solvd/laba/db/util/ConfigFileDAO.java:
```diff
@@ -0,0 +1,35 @@
+            dataSource.setMaxIdle(10);
+            dataSource.setMaxWaitMillis(200);
+        } catch (IOException e) {
+            logger.info("Error loading the property file");
```
It is better to log the IOException as an error

jishaGithub replied:
Yes. you are right. I have updated my code.

adelagega commented on DbConnection/src/main/java/com/solvd/laba/db/util/ConfigFileDAO.java:
```diff
@@ -0,0 +1,35 @@
+    public static void loadPropertyConfigFile() {
+        Properties property = new Properties();
+        try (InputStream dbPropertiesStream = ConfigFileDAO.class.getClassLoader().getResourceAsStream("db.properties")) {
+            property.load(dbPropertiesStream);
```
you are not checking if dbPropertiesStream is null before calling property.load(dbPropertiesStream)

jishaGithub replied:
Done

adelagega commented on DbConnection/src/main/java/com/solvd/laba/db/dao/BranchDAO.java:
```diff
@@ -0,0 +1,101 @@
+            }
+            logger.info("No record found. Invalid ID");
+        } catch (SQLException e) {
+            logger.info("SQL query error:"+e.getMessage());
```
You should use logger.error to log exceptions

jishaGithub replied:
Done

adelagega commented on DbConnection/src/main/java/com/solvd/laba/db/service/BranchLocationService.java:
```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+
+public class BranchLocationService implements GenericDAO<BranchLocation> {
+    private static final Logger logger = LogManager.getLogger(BranchLocationService.class);
```
This is not used

jishaGithub replied:
Removed unused variables

jishaGithub committed:
Update ConfigFileDAO.java

jishaGithub committed:
Update ConfigFileDAO.java

jishaGithub committed:
Update ConfigFileDAO.java

jishaGithub committed:
Update BranchLocationService.java

jishaGithub committed:
Update BranchService.java

jishaGithub committed:
Update CustomerAuthenticationService.java

jishaGithub committed:
Update CustomerService.java

jishaGithub committed:
Update EmployeeService.java

jishaGithub committed:
Update InsuranceService.java

jishaGithub committed:
Update PaymentMethodService.java

jishaGithub committed:
Update PaymentService.java

jishaGithub committed:
Update PromotionService.java

jishaGithub committed:
Update RentalRateService.java

jishaGithub committed:
Update RentalService.java

jishaGithub committed:
Update VehicleCategoryService.java

jishaGithub committed:
Update VehicleFeatureService.java

jishaGithub committed:
Update VehicleHasVehicleFeatureService.java

jishaGithub committed:
Update VehicleService.java

jishaGithub committed:
Updated logger.info to logger.error

jishaGithub committed:
Fixed logger.info to logger.error in dao classes
